xarray/test/test_dataarray.py

```diff
@@ -2268,3 +2268,38 @@ def test_dot(self):
             da.dot(dm.values)
         with self.assertRaisesRegexp(ValueError, 'no shared dimensions'):
             da.dot(DataArray(1))
+
+    def test_binary_op_join_setting(self):
+        """
+        A test method to verify the ability to set binary operation join kwarg
+        ("inner", "outer", "left", "right") via xr.set_options().
+        """
+        # First we set up a data array
+        xdim, ydim, zdim = 'x', 'y', 'z'
+        xcoords, ycoords, zcoords = ['a', 'b', 'c'], [-2, 0, 2], [0, 1, 2]
+        total_size = len(xcoords) * len(ycoords) * len(zcoords)
+        # create a 3-by-3-by-3 data array
+        arr = xr.DataArray(np.arange(total_size).\
+                           reshape(len(xcoords),len(ycoords),len(zcoords)),
+                           [(xdim, xcoords), (ydim, ycoords),(zdim, zcoords)])
+        # now create a data array with the last x slice missing
+        arr1 = arr[0:-1,:,:].copy()
+        # create another data array with the last z slice missing
+        arr2 = arr[:,:,0:-1].copy()
+        # because the default in OPTIONS is join="inner", we test "outer" first
+        xr.set_options(join="outer")
+        result = arr1 + arr2
+        self.assertTrue(result.size == total_size) # should be 3 * 3 * 3
+        self.assertTrue(result.shape == arr.shape)
+        self.assertTrue(result[-1,:,:].isnull().all())
+        self.assertTrue(result[:,:,-1].isnull().all())
+        # now revert back to join="inner"
+        xr.set_options(join="inner")
```

Comment:
Even though it'd mean duplicating some setup code, I would split this larger case into two functions since you're testing two join types. I would say you could make a small class to avoid the duplication, but it looks like pydata has gone with the one-class-per-file approach, at least for these tests. https://docs.python.org/2/library/unittest.html#unittest.TestCase funny given first line